### PR TITLE
Update tray icon with vpn state change

### PR DESCRIPTION
**Pull request by spylogsster**
Resolves https://github.com/brave/brave-browser/issues/32535
Resolves https://github.com/brave/brave-browser/issues/32534

- Synced states for wireguard status tray icon and browser vpn state. Browser could be started/stopped independetly from wireguard tunnel service. In order to reflect actual connection state Tunnel service status has higher priority over data fetched from the browser only for connected state.
- Moved utility functions from tunnel_utils to storage_utils because they are working with storage only.
- Refactored status tray state machine
- Added tests for code around service storage
- Updated icons with new Figma design

https://github.com/brave/brave-core/assets/2965009/cc7fe6d6-345d-42be-960a-20787abcca11

## Submitter Checklist:

- [x] I confirm that no [security/privacy review is needed](https://github.com/brave/brave-browser/wiki/Security-reviews) and no other type of reviews are needed, or that I have [requested](https://github.com/brave/reviews/issues/new/choose) them
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [x] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [x] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

- STR from the issues

### CODE REVIEW

File: browser/brave_vpn/win/brave_vpn_wireguard_service/status_tray/status_tray_runner.cc [hunk 1]
```diff
@@ -164,22 +182,62 @@ void StatusTrayRunner::OnConnected(bool success) {
...
+    // Check if we have obsolete connected state in storage.
+    state = IsTunnelServiceRunning()
+                ? state
+                : brave_vpn::mojom::ConnectionState::DISCONNECTED;
+    // if Tunnel service launched it means we have connected state and should
+    // reset storage states because it could be expired from closed browser.
+    wireguard::WriteConnectionState(static_cast<int>(state));
+  }
+
+  if (current_state_.has_value() && current_state_.value() == state) {
```
Comment (sangwoo108) [inline browser/brave_vpn/win/brave_vpn_wireguard_service/status_tray/status_tray_runner.cc:215]: totally optional: I think absl::optional's `operater==` covers has_value() so could be simplified

```suggestion
  if (current_state_ == state) {
```

File: browser/brave_vpn/win/brave_vpn_wireguard_service/status_tray/status_tray_runner.h [hunk 2]
```diff
@@ -37,29 +42,44 @@ class StatusTrayRunner : public TrayMenuModel::Delegate {
...
   void SubscribeForServiceStopNotifications(const std::wstring& name);
   void SubscribeForStorageUpdates();
   void OnConnected(bool success);
   void OnDisconnected(bool success);
   void OnServiceStateChanged(int mask);
   void OnStorageUpdated();
-
+  bool IsIconCreated();
+  void SetIconState(int icon_id, int tooltip_id);
+  brave_vpn::mojom::ConnectionState GetConnectionState();
```
Comment (sangwoo108) [inline browser/brave_vpn/win/brave_vpn_wireguard_service/status_tray/status_tray_runner.h:76]: nit:

```suggestion
  brave_vpn::mojom::ConnectionState GetConnectionState();
  
```

File: components/brave_vpn/common/wireguard/win/storage_utils_unittest.cc [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+#include "brave/components/brave_vpn/common/wireguard/win/storage_utils.h"
+#include "base/test/test_reg_util_win.h"
+#include "base/win/registry.h"
+#include "brave/components/brave_vpn/common/wireguard/win/wireguard_utils.h"
+#include "testing/gmock/include/gmock/gmock.h"
+#include "testing/gtest/include/gtest/gtest.h"
+
+namespace brave_vpn {
+
```
Comment (sangwoo108) [inline components/brave_vpn/common/wireguard/win/storage_utils_unittest.cc:14]: nit: for pairing with the closing namespaces

```suggestion
```

**Review by sangwoo108 — APPROVED** [review]
LGTM 👍

**Review by mkarolin — APPROVED** [review]
`strings`++

**Pull request merged**